src/App.tsx

```diff
@@ -1,61 +1,102 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TODO } from './types/todo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
 export const App = () => {
+  const [visibleTodos, setVisibleTodos] = useState([...todosFromServer]);
+
+  const [title, setTitle] = useState('');
+  const [currUser, setCurrUser] = useState(0);
+
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setCurrUser(0);
+  };
+
+  const submit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (currUser === 0) {
+      setUserError(true);
+    }
+
+    if (!title || !currUser) {
+      return;
+    }
+
+    const todo: TODO = {
+      id: [...visibleTodos].sort((a, b) => b.id - a.id)[0].id + 1 || 1,
```

Comment:
The logic for generating a new todo ID assumes that `visibleTodos` is never empty. If `visibleTodos` is empty, accessing `[0].id` will result in an error. Consider checking if `visibleTodos` is empty before trying to access the first element.